src/LineageTree/lineageTreeManager.py

```diff
@@ -0,0 +1,177 @@
+from functools import partial
+
+import numpy as np
+from edist.uted import uted
+
+from LineageTree import lineageTree
+
+
+class LineageTreeManager:
+    def __init__(self):
+        self.lineagetrees = {}
+        self.classification = {"Wt": {}, "Ptb": {}}
+        self.lineageTree_counter = 0
+        self.registered = {}
+
+    def get_next_tree(self):
+        self.lineageTree_counter += 1
+        return self.lineageTree_counter - 1
+
+    def add(
+        self, other_tree: lineageTree, name: str = "", classification: str = ""
+    ):
+        """Function that adds a new lineagetree object to the class.
+        Can be added either by .add or by using the + operator. If a name is
+        specified it will also add it as this specific name, otherwise it will
+        use the already existing name of the lineagetree. Lastly, it adds it to
+        the classification dict.
+
+        Args:
+            other_tree (lineageTree): Thelineagetree to be added.
+            name (str, optional): Then name of. Defaults to "".
+            classification (str, optional): If the user need to add it to classification
+                                    provide here the name of the class.
+                                    Defaults to "".
+
+        Returns:
+            _type_: _description_
+        """
+        for tree in self.lineagetrees.values():
+            if tree == other_tree:
+                return False
+        if isinstance(other_tree, lineageTree):
+            if name:
+                self.lineagetrees[name] = other_tree
+            else:
+                try:
+                    name = other_tree.name
+                    self.lineagetrees[name] = other_tree
+                except:
+                    self.lineagetrees[
+                        f"Lineagetree {self.get_next_tree()}"
+                    ] = other_tree
+        # if classification in ("Wt", "Ptb"):
+        #     self.classification[type] = {name: other_tree}
+
+    def __add__(self, other):
+        self.add(other)
+
+    def classify_existing(self, key, classification: str):
+        if classification in ("Wt", "Ptb"):
+            self.classification[classification] = {key: self.lineagetrees[key]}
+        else:
+            return False
+
+    def write(self, fname: str):
+        """Saves the manager
+
+        Args:
+            fname (str): The path and name of the file that is to be saved.
+        """
+        if os.path.splitext(fname)[-1] != ".ltM":
+            fname = os.path.extsep.join((fname, "ltM"))
+        with open(fname, "bw") as f:
+            pkl.dump(self, f)
+            f.close()
+
+    def remove_embryo(self, key):
+        """Removes the embryo from the manager.
+
+        Args:
+            key (str): The name of the lineagetree to be removed
+
+        Raises:
+            Exception: If there is not such a lineagetree
+        """
+        if key in self.lineagetrees:
+            del self.lineagetrees[key]
```

Comment:
you should not have to use the keyword `del`.
I think you'd rather want to use `pop`:
```python
self.lineagetrees.pop(key, None)
self.classification["Wt"].pop(key, None)
self.classification["Ptb"].pop(key, None)
```
You don't need to check if the key is there.